lib/jira/client/index.js

```diff
@@ -107,13 +132,24 @@ module.exports = async (id, installationId, jiraHost) => {
         delete: (repositoryId) => instance.delete(`/rest/devinfo/0.10/repository/${repositoryId}`, {
           fields: { _updateSequenceId: Date.now() }
         }),
-        update: (data, options) => instance.post('/rest/devinfo/0.10/bulk', {
-          preventTransitions: (options && options.preventTransitions) || false,
-          repositories: [data],
-          properties: {
-            installationId
+        update: (data, options) => {
+          // Deduplicate issue keys in commits and branches before request
+          if ('commits' in data) data.commits = dedupIssueKeys(data.commits)
+          if ('branches' in data) data.branches = dedupIssueKeys(data.branches)
```

Comment:
Is it possible for `branches[0].lastCommit.issueKeys` to exceed the issue key limit?

Reply:
Great catch! Just added a check for that here: https://github.com/integrations/jira/blob/df5547f6ab8075425b28386e241cec6a48e8b7e7/lib/jira/client/index.js#L140-L144